Solutions/Corvus.Azure.Storage.Tenancy/Corvus/Azure/Storage/Tenancy/TableStorageTenantExtensions.cs

```diff
@@ -0,0 +1,126 @@
+﻿// <copyright file="TableStorageTenantExtensions.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Corvus.Azure.Storage.Tenancy
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Corvus.Tenancy;
+    using Newtonsoft.Json.Linq;
+
+    /// <summary>
+    /// Extension methods for storage features for the tenant.
+    /// </summary>
+    public static class TableStorageTenantExtensions
+    {
+        /// <summary>
+        /// Determines whether the tenant has a table storage configuration.
+        /// </summary>
+        /// <param name="tenant">The tenant.</param>
+        /// <param name="definition">The table storage container definition.</param>
+        /// <returns>True if there is an explicit configuration set for this table storage container definition.</returns>
+        /// <remarks>This can be more efficient than trying to get the property, as it avoids a deserialization.</remarks>
+        public static bool HasStorageTableConfiguration(this ITenant tenant, TableStorageTableDefinition definition)
+        {
+            if (tenant is null)
+            {
+                throw new ArgumentNullException(nameof(tenant));
+            }
+
+            if (definition is null)
+            {
+                throw new ArgumentNullException(nameof(definition));
+            }
+
+            return ((JObject)tenant.Properties)[GetConfigurationKey(definition)] != null;
```

Comment:
Have we taken a dependency on the Json.NET-specific property bag bits? If not, should we really be casting to `JObject`?

Reply:
Looks like this was never updated on the blob storage equivalent when the property bag changes were made.